pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2020 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+)
+
+// Important:
+// The procedure working with this file:
+// 1. Edit the file
+// 1. Run "make generate" to regenerate code
+// 2. Run "make manifests" to regenerate the CRD
+
+func init() {
+	SchemeBuilder.Register(&AtlasProject{}, &AtlasProjectList{})
+}
+
+type ScopeType string
+
+const (
+	ClusterScopeType  ScopeType = "CLUSTER"
+	DataLakeScopeType ScopeType = "DATA_LAKE"
+)
+
+// AtlasDatabaseUserSpec defines the desired state of Database User in Atlas
+type AtlasDatabaseUserSpec struct {
+	// Project is a reference to AtlasProject resource the user belongs to
+	Project ResourceRef `json:"projectRef"`
+
+	// DatabaseName is a Database against which Atlas authenticates the user. Must be equal to 'admin'
+	// as only SCRAM authentication is supported so far.
+	DatabaseName string `json:"databaseName"`
```

Comment:
can this be hardcoded & not appear in the CRD until we support more auth forms?

Reply:
let me check if I understand you correctly: do you want to remove the comment from CRD? Because as soon as we support other auth forms - the comment won't be actual anymore, it's only for the Trial version of the Operator.

Reply:
I would drop this field from the CRD entirely -- and then add it when there are values that the user should specify

I don't think it's good design to have a field where the user must type the string "admin" -- it makes work for them , without any benefit.

Reply:
Alternatively, we could add a default value via `kubebuilder`, like `// +kubebuilder:default=admin`

Reply:
@jamesbroadhead I see your point. Though I have concerns about omitting this field now because when we make this mandatory in the future when we start supporting more auth modes - this will break the existing CRs for the users and they will have to modify the specifications to fix tjos. I'd rather made this field mandatory from the beginning or used the default value suggested by Pavel

Reply:
defaulting to `admin` sounds like a good idea here, I don't think it's a good user experience to need to specify the same value no matter what.